yt_dlp/extractor/lecturio.py

```diff
@@ -167,8 +159,20 @@ def _real_extract(self, url):
             'automatic_captions': automatic_captions,
         }
 
+# German Lecturio simply requires different URLs
+class LecturioDeIE(LecturioIE):
```

Comment:
Don't subclass from a concrete IE

Add back `LecturioBaseIE`, and move all code from `LecturioIE` to it *except* `_VALID_URL`, `_TESTS`.

Add a new `LecturioDeBaseIE`, subclassed from `LecturioBaseIE`, that redefines `_API_BASE_URL` and `_LOGIN_URL`

Subclass `LecturioIE` and `LecturioCourseIE` from `LecturioBaseIE`

Subclass `LecturioDeIE` and `LecturioDeCourseIE` from `LecturioDeBaseIE`